src/pages/index.js

```diff
@@ -0,0 +1,299 @@
+};
+
+function handleLikeClick(likeBtn, cardId) {
+  const isLiked = likeBtn.classList.contains("card__like-btn_active");
+  api
+    .changeLikeStatus(cardId, !isLiked)
+    .then((card) => {
+      const isNowLiked = card.isLiked;
+      likeBtn.classList.toggle("card__like-btn_active", isNowLiked);
+    })
+    .catch((err) => console.error("Error changing like status:", err));
+}
+
+function getCardElement(data) {
+  const cardElement = dom.cardTemplate.cloneNode(true);
+  cardElement.querySelector(".card__title").textContent = data.name;
+  const cardImage = cardElement.querySelector(".card__image");
+  cardImage.alt = data.name;
+  cardImage.src = data.link;
+  const likeBtn = cardElement.querySelector(".card__like-btn");
+  const isLiked = (data.likes || []).some((user) => user._id === currentUserId);
+  likeBtn.classList.toggle("card__like-btn_active", isLiked);
+  if (data.isLiked) {
+    likeBtn.classList.add("card__like-btn_active");
+  } else {
+    likeBtn.classList.remove("card__like-btn_active");
+  }
+  likeBtn.addEventListener("click", () => handleLikeClick(likeBtn, data._id));
+  cardElement.querySelector(".card__del-btn").addEventListener("click", () => handleDelCard(cardElement, data));
+  cardImage.addEventListener("click", () => openPreviewModal(data));
+  return cardElement;
+}
+
+function renderCard(data, container) {
+  const cardElement = getCardElement(data);
+  container.prepend(cardElement);
+}
+
+function openPreviewModal(data) {
+  dom.previewImage.src = data.link;
+  dom.previewImage.alt = data.name;
+  dom.previewCaption.textContent = data.name;
+  openModal(dom.previewModal);
+}
+
+function handleDelCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(dom.deletePostModal);
+}
+
+function handleDelSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(dom.submitDeletePostBtn, "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(dom.deletePostModal);
+      selectedCard = null;
+      selectedCardId = null;
+    })
+    .catch((err) => console.error("Error deleting card:", err))
+    .finally(() => {
+      setButtonText(dom.submitDeletePostBtn, "Delete");
+      disableBtn(dom.submitDeletePostBtn, settings);
+    });
+}
+dom.submitDeletePostBtn.addEventListener("click", handleDelSubmit);
+
+function handleAvatarEditSave(evt) {
+  evt.preventDefault();
+  setButtonText(dom.avatarEditSaveBtn, "Saving...");
+  api
+    .editAvatar({ avatar: dom.avatarEditInput.value })
+    .then((user) => {
+      dom.profileImg.src = user.avatar;
+      closeModal(dom.avatarEditModal);
+      dom.avatarEditForm.reset();
+      disableBtn(dom.avatarEditSaveBtn, settings);
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      setButtonText(dom.avatarEditSaveBtn, "Save");
+      disableBtn(dom.avatarEditSaveBtn, settings);
```

Comment:
`      disableBtn(dom.avatarEditSaveBtn, settings);
`

Please, delete it because you disable the button in line 199 in `.then`. It's enough there

You need to **disable** the button only in a `.then` block after a **successful response** to be able to click it again if there is a server error.